src/storage/src/hummock/local_version/local_version_manager.rs

```diff
@@ -113,25 +114,32 @@ pub struct LocalVersionManager {
 
 impl LocalVersionManager {
     #[allow(clippy::new_ret_no_self)]
-    pub fn new(
+    #[allow(clippy::too_many_arguments)]
+    pub async fn new(
         options: Arc<StorageConfig>,
         sstable_store: SstableStoreRef,
         stats: Arc<StateStoreMetrics>,
         hummock_meta_client: Arc<dyn HummockMetaClient>,
         write_conflict_detector: Option<Arc<ConflictDetector>>,
         sstable_id_manager: SstableIdManagerRef,
         filter_key_extractor_manager: FilterKeyExtractorManagerRef,
+        mut version_update_rx: mpsc::UnboundedReceiver<pin_version_response::Payload>,
```

Comment:
Could we pass the whole `PinnedVersion` as argument to avoid this method become `async` ?